plugins/Web/WebBackend.vala

```diff
@@ -0,0 +1,176 @@
+// -*- Mode: vala; indent-tabs-mode: nil; tab-width: 4 -*-
+/*-
+ * Copyright (c) 2013 Maya Developers (https://launchpad.net/maya)
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ * Authored by: Corentin Noël <tintou@mailoo.org>
+ */
+
+public Maya.Backend get_backend (Module module) {
+    debug ("Activating Web Backend");
+    var b = new Maya.WebBackend ();
+    b.ref ();
+    return b;
+}
+
+public static Maya.Backend backend;
+
+public class Maya.WebBackend : GLib.Object, Maya.Backend {
+
+    public WebBackend () {
+        backend = this;
+    }
+
+    public string get_name () {
+        return _("On the web");
+    }
+
+    public string get_uid () {
+        return "webcal-stub";
+    }
+
+    public Gee.Collection<PlacementWidget> get_new_calendar_widget (E.Source? to_edit = null) {
+        var collection = new Gee.LinkedList<PlacementWidget> ();
+
+        bool keep_copy = false;
+        if (to_edit != null) {
+            E.SourceOffline source_offline = (E.SourceOffline)to_edit.get_extension (E.SOURCE_EXTENSION_OFFLINE);
+            keep_copy = source_offline.stay_synchronized;
+        }
+
+        collection.add (Maya.DefaultPlacementWidgets.get_keep_copy (0, keep_copy));
+
+        var url_label = new PlacementWidget ();
+        url_label.widget = new Gtk.Label (_("URL:"));
+        ((Gtk.Label) url_label.widget).expand = true;
```

Comment:
In preparation for GTK4 using hexpand and vexpand instead of expand is preferred.